src/NuGet.Clients/NuGet.PackageManagement.UI/Models/Package/DeprecationCapability.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Linq;
+using NuGet.Protocol.Model;
+using NuGet.VisualStudio.Internal.Contracts;
+
+namespace NuGet.PackageManagement.UI
+{
+    public class DeprecationCapability : IDeprecation
+    {
+        public DeprecationCapability(PackageDeprecationMetadataContextInfo deprecatedInfo)
+        {
+            DeprecationMetadata = deprecatedInfo;
+        }
+
+        public PackageDeprecationMetadataContextInfo DeprecationMetadata { get; private set; }
```

Comment:
```suggestion
        public PackageDeprecationMetadataContextInfo DeprecationMetadata { get; }
```
I don't think you need the setter if you're only setting it in the constructor.